src/expr/src/linear.rs

```diff
@@ -1703,6 +1703,18 @@ pub mod plan {
             }
         }
 
+        /// Returns an iterator over mutable references to all non-temporal
+        /// scalar expressions in the plan.
+        ///
+        /// The order of iteration is unspecified.
+        pub fn iter_nontemporal_exprs(&mut self) -> impl Iterator<Item = &mut MirScalarExpr> {
```

Comment:
Kind of a gross public function to have to expose, but it does the job. Open to suggestions.

Reply:
Visitor that takes a closure that takes the current item and return whether it should be visited?